File: app/src/main/java/com/yanxing/ui/DesignFragment.java
package com.yanxing.ui;

import android.graphics.Color;
import android.support.design.widget.CollapsingToolbarLayout;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.yanxing.adapterlibrary.RecyclerViewAdapter;
import com.yanxing.base.BaseFragment;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;

/**
 * Design中控件用法
 * Created by lishuangxiang on 2016/11/18.
 */

public class DesignFragment extends BaseFragment {

    @BindView(R.id.collapsing_toolbar_layout)
    CollapsingToolbarLayout mCollapsingToolbarLayout;

    @BindView(R.id.recyclerView)
    RecyclerView mRecyclerView;

    private RecyclerViewAdapter<String> mRecyclerViewAdapter;

    @Override
    protected int getLayoutResID() {
        return R.layout.fragment_design;
    }

    @Override
    protected void afterInstanceView() {
        mCollapsingToolbarLayout.setTitle("CollapsingToolbarLayout");
        mCollapsingToolbarLayout.setExpandedTitleColor(Color.WHITE);
        mCollapsingToolbarLayout.setCollapsedTitleTextColor(Color.GREEN);

        List<String> strings = new ArrayList<>();
        for (int i = 0; i < 80; i++) {
            strings.add(String.valueOf(i));
        }
        mRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        mRecyclerViewAdapter = new RecyclerViewAdapter<String>(strings, R.layout.adapter_recycler_view) {

            @Override
            public void onBindViewHolder(RecyclerViewAdapter.MyViewHolder holder, final int position) {
                holder.setText(R.id.text,mDataList.get(position));
            }
        };
        mRecyclerView.setAdapter(mRecyclerViewAdapter);
    }
}

File: app/src/main/java/com/yanxing/ui/DesignFragment.kt
package com.yanxing.ui

import android.graphics.Color
import androidx.recyclerview.widget.LinearLayoutManager

import com.yanxing.base.BaseFragment
import com.yanxing.commonlibrary.adapter.RecyclerViewAdapter

import java.util.ArrayList

import kotlinx.android.synthetic.main.fragment_design.*

/**
 * Design中控件用法
 * Created by lishuangxiang on 2016/11/18.
 */
class DesignFragment : BaseFragment() {

    override fun getLayoutResID(): Int {
        return R.layout.fragment_design
    }

    override fun afterInstanceView() {
        collapsingToolbarLayout.title = "CollapsingToolbarLayout"
        collapsingToolbarLayout.setExpandedTitleColor(Color.WHITE)
        collapsingToolbarLayout.setCollapsedTitleTextColor(Color.GREEN)

        val strings = ArrayList<String>()
        for (i in 0..79) {
            strings.add(i.toString())
        }
        recyclerView.layoutManager = androidx.recyclerview.widget.LinearLayoutManager(activity)
        val recyclerViewAdapter = object : RecyclerViewAdapter<String>(strings, R.layout.adapter_recycler_view) {

            override fun onBindViewHolder(holder: RecyclerViewAdapter.MyViewHolder, position: Int) {
                holder.setText(R.id.text, mDataList[position])
            }
        }
        recyclerView.adapter = recyclerViewAdapter
    }
}
